draft-packages/menu/KaizenDraft/Menu/MenuDropdown.tsx

```diff
@@ -15,126 +15,102 @@ type MenuDropdownProps = {
   align?: "left" | "right"
   width?: "default" | "contain"
   autoHide?: "on" | "outside-click-only" | "off"
+  children: React.ReactNode
+  referenceElement: HTMLElement | null
 }
 
-class MenuDropdown extends React.Component<MenuDropdownProps> {
-  static displayName = "MenuDropdown"
-
-  static defaultProps = {
-    autoHide: "on",
-  }
-
-  menu = React.createRef<HTMLDivElement>()
-
-  componentDidMount() {
-    const { autoHide } = this.props
-    if (autoHide !== "off") {
-      document.addEventListener(
-        "click",
-        this.handleDocumentClickForAutoHide,
-        false
-      )
-    }
-    window.addEventListener("resize", this.handleDocumentResize, false)
-    this.positionMenu()
-  }
-
-  componentWillUnmount() {
-    const { autoHide } = this.props
-    if (autoHide !== "off") {
-      document.removeEventListener(
-        "click",
-        this.handleDocumentClickForAutoHide,
-        false
-      )
-    }
-    window.removeEventListener("resize", this.handleDocumentResize, false)
-  }
-
-  componentWillUpdate(newProps) {
-    // Hm, I don't like hooks, but in this situation they would have been handy
-    if (this.props.autoHide === "off" && newProps.autoHide !== "off") {
-      document.addEventListener(
-        "click",
-        this.handleDocumentClickForAutoHide,
-        false
-      )
-    } else if (this.props.autoHide !== "off" && newProps.autoHide === "off") {
-      document.removeEventListener(
-        "click",
-        this.handleDocumentClickForAutoHide,
-        false
-      )
-    }
-  }
-
-  positionMenu() {
-    const menu = this.menu
-
-    if (!this.props.position || !menu) {
-      return
+const MenuDropdown = ({
+  children,
+  referenceElement,
+  id,
+  hideMenuDropdown,
+  autoHide = "on",
+  align = "left",
+  width = "default",
+}: MenuDropdownProps) => {
+  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
+    null
+  )
+  const { styles: popperStyles, attributes } = usePopper(
+    referenceElement,
+    popperElement,
+    {
+      modifiers: [
+        {
+          name: "offset",
+          options: {
+            offset: [0, 6], // value used from the $kz-spacing-xs scss variable
```

Comment:
/meganit and you don't need to make this change. You can still use the tokens if you need them in javascript, like so: 

```
import { spacing } from "@kaizen/design-tokens/tokens/spacing"

options: {
   offset: [0, parseInt(spacing.kz.spacing.xs, 10)], 
 },
```